pkg/api/v1/atlasbackupcompliancepolicy_types.go

```diff
@@ -0,0 +1,158 @@
+package v1
+
+import (
+	"strings"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/status"
+)
+
+var _ api.AtlasCustomResource = &AtlasBackupCompliancePolicy{}
```

Comment:
what is this? was it manual or automatic?

Reply:
this is a compile time assertion whether `&AtlasBackupCompliancePolicy{}` implements the `api.AtlasCustomResource` interface correctly. If not, the compiler will shout out you.

Reply:
This is duplicating here https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1623/files#diff-6de4adb50a8a05768b2b2e278432a2fa7690403581a98cfbaba1aa8118ae0ae7R31

Reply:
Removed this and left the one @helderjs referenced.